packages/deploy/README.md

```diff
@@ -0,0 +1,131 @@
+Start by creating a new _Team API Key_ in Defender, and granting it the capability to manage sentinels. Use the newly created API key to initialize an instance of the Sentinel client.
+
+```js
+const { PlatformClient } = require('platform-deploy-client');
+const client = PlatformClient({ apiKey: API_KEY, apiSecret: API_SECRET });
+```
+
+### Deployment Config
+
+In order to deploy a contract you will need to create a deployment config. The config consists of the ID belonging to the Relayer you want to use for deployments. You can only specify one deployment config per network, so if you create a config with a Relayer on the `goerli` testnet all you your deployments on `goerli` will be sent via that relayer.
+
+```js
+await client.DeploymentConfig.create({ relayerId: 'dfa8b9a6-0f88-4d28-892a-93e1f5a8d2a7' });
+```
+
+You can also list your deployment configs, which will return a `DeploymentConfigResponse[]` object
+
+```js
+await client.DeploymentConfig.list();
+```
+
+As well as fetching a config via it's ID
+
+```js
+const deploymentConfigId = 'e595ce88-f525-4d5d-b4b9-8e859310b6fb';
+await client.DeploymentConfig.get(deploymentConfigId);
+```
+
+If you want to change the relayer used in a deploy config you can update it
+
+```js
+const deploymentConfigId = 'e595ce88-f525-4d5d-b4b9-8e859310b6fb';
+await client.DeploymentConfig.update(deploymentConfigId, { relayerId: '3dcfee82-f5bd-43e3-8480-0676e5c28964' });
+```
+
+### Deployment
+
+To deploy a contract you need to provide these required fields:
+
+- `network`
+- `contractName`
+- `contractPath` - The path of your contract in your hardhat project
+
+Additionally you must provide your compilation artifact from hardhat. The compilation artifact can be found in your hardhat project folder at `artifacts/build-info/{build-id}.json`. Either one of these fields are required:
+
+- `artifactPayload` - JSON stringified version of the file
+- `artifactUri` - URI to the hosted artifact file
+
+There are a number of optional fields depending on what you are deploying, these include:
+
+- `constructorInputs` - The inputs to your contract constructor,
+- `value` - ETH to be sent with the deployment
+- `salt` - deployments are done using the CREATE2 opcode, you can provide a salt or we can generate one for you if none is supplied
+- `licenseType` - This will be displayed on Etherscan e.g MIT
+- `libraries` - If you contract uses any external libraries they will need to be added here in the format `{ [LibraryName]: LibraryAddress }`
+
+Below is an example of a contract deployment request which responds with a `DeploymentResponse`
+
+```js
+await client.Deployment.deploy({
+  contractName: 'Greeter',
+  contractPath: 'contracts/Greeter.sol',
+  network: 'goerli',
+  artifactPayload: JSON.stringify(artifactFile),
+  licenseType: 'MIT',
+  constructorInputs: ['Hello World!'],
+});
+```
+
+You can also list your deployments, which will return a `DeploymentResponse[]` object
+
+```js
+await client.Deployment.list();
```

Comment:
Are we returning all deployments at once? Thinking that in case of a lot deployments pagination with next cursor would be nice.